dashboard/src/actions/profileActions.js

```diff
@@ -0,0 +1,107 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+import * as API_ROUTES from "../utils/apiConstants";
+import Cookies from "js-cookie";
+import { showFailureToast } from "./toastActions";
+
+export const getProfileDetails = () => async (dispatch) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const username = Cookies.get("username") ? Cookies.get("username") : null;
```

Comment:
Is it necessary to call `Cookies.get` twice here? According to the documentation, it seems to just return `novalue` if the cookie doesn't exist.

Also, I suspect that if we don't have a username here, the attempt to show a profile should just fail; the `/api/v1/user/null` isn't going to do anything useful anyway.